{
    "active_lock_reason": "too heated",
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "<!-- DO NOT DELETE \r\nvalidate_template=true\r\ntemplate_path=.github/ISSUE_TEMPLATE/bug.md\r\n-->\r\n\r\n### [REQUIRED] Step 2: Describe your environment\r\n\r\n  * Firebase Component: Database\r\n  * Component version: 19.2.0, but I think it has been problem since 2018\r\n\r\n### [REQUIRED] Step 3: Describe the problem\r\n\r\nUsing realtime database change listener normally in android mobile with wifi connection. It work perfectly for most of the times. But it would eventually (2-3 days to 2-3 weeks) not be able to connect to firebase realtime database and get the realtime update, even if the PC of the same router still able to connect. I need to shut the wifi of my android to use mobile 4g network, or reset the router. Resetting router make it usable but eventually after a while it will fail again\r\n\r\nI myself using unity. But there was already some people happen to face the same problem so I think it could be common to many people\r\n\r\nhttps://stackoverflow.com/questions/52609349/couldnt-connect-to-firebase-database-via-wifi-but-connects-fine-with-4g-mobile\r\nhttps://stackoverflow.com/questions/48340327/firebase-database-listeners-dont-work-on-android-with-wifi\r\n\r\nAlso, surprisingly, the web that use firebase still work in the same network state. So I think this problem is because native network library\r\n\r\nI think the actual source of problem could be a problem of router setting. But that might be critical. Because realtime update could be fail for anyone in misconfig router (in workplace for instance) but it result in the whole app failed unknowingly to us. It could also be a critical feature of that app. Isn't it possible to have firebase native to have failsafe to fallback?",
    "closed_at": null,
    "closed_by": null,
    "comments": 21,
    "comments_url": "https://api.github.com/repos/firebase/firebase-android-sdk/issues/1258/comments",
    "created_at": "2020-02-20T10:58:17Z",
    "events_url": "https://api.github.com/repos/firebase/firebase-android-sdk/issues/1258/events",
    "html_url": "https://github.com/firebase/firebase-android-sdk/issues/1258",
    "id": 568217585,
    "labels": [
        {
            "color": "d73a4a",
            "default": false,
            "description": "Something isn't working",
            "id": 1041939859,
            "name": "type: bug",
            "node_id": "MDU6TGFiZWwxMDQxOTM5ODU5",
            "url": "https://api.github.com/repos/firebase/firebase-android-sdk/labels/type:%20bug"
        },
        {
            "color": "f8f8f8",
            "default": false,
            "description": null,
            "id": 1258595136,
            "name": "api: database",
            "node_id": "MDU6TGFiZWwxMjU4NTk1MTM2",
            "url": "https://api.github.com/repos/firebase/firebase-android-sdk/labels/api:%20database"
        }
    ],
    "labels_url": "https://api.github.com/repos/firebase/firebase-android-sdk/issues/1258/labels{/name}",
    "locked": true,
    "milestone": null,
    "node_id": "MDU6SXNzdWU1NjgyMTc1ODU=",
    "number": 1258,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/firebase/firebase-android-sdk/issues/1258/reactions"
    },
    "repository_url": "https://api.github.com/repos/firebase/firebase-android-sdk",
    "state": "open",
    "state_reason": null,
    "timeline_url": "https://api.github.com/repos/firebase/firebase-android-sdk/issues/1258/timeline",
    "title": "Firebase Realtime Database don't get update from wifi",
    "updated_at": "2021-06-10T17:40:43Z",
    "url": "https://api.github.com/repos/firebase/firebase-android-sdk/issues/1258",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/1042507?v=4",
        "events_url": "https://api.github.com/users/Thaina/events{/privacy}",
        "followers_url": "https://api.github.com/users/Thaina/followers",
        "following_url": "https://api.github.com/users/Thaina/following{/other_user}",
        "gists_url": "https://api.github.com/users/Thaina/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/Thaina",
        "id": 1042507,
        "login": "Thaina",
        "node_id": "MDQ6VXNlcjEwNDI1MDc=",
        "organizations_url": "https://api.github.com/users/Thaina/orgs",
        "received_events_url": "https://api.github.com/users/Thaina/received_events",
        "repos_url": "https://api.github.com/users/Thaina/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/Thaina/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/Thaina/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/Thaina"
    }
}